### DISCUSSION TITLE
Custom Auxiliary Functions for SQLite

### DISCUSSION THREAD

**Discussion by Boscop** [diesel-rs/diesel#2763] [Ideas]
Similar to SQLite custom functions (https://github.com/diesel-rs/diesel/issues/1103) which diesel now supports (https://github.com/diesel-rs/diesel/pull/1691), SQLite also has the related concept of [Custom Auxiliary Functions for its full-text search (FTS) functionality](https://sqlite.org/fts5.html#custom_auxiliary_functions).
> Implementing a custom auxiliary function is similar to implementing a scalar SQL function.

It would be useful to be able to register a Rust function with sqlite in the same way as normal sqlite custom functions.

**Comment by weiznich** [comment]
We are happy to receive a concrete proposal for such a api + some thoughts about how to handle this for sqlite distributions with FTS disabled. Otherwise this is at least for me nothing I plan to work on for a foreseeable future.
